src/main/java/com/vmware/vipclient/i18n/VIPCfg.java

```diff
@@ -117,17 +125,46 @@ public void initialize(String vipServer, String productName, String version) {
         this.version = version;
         this.vipServer = vipServer;
     }
-
+    
+    /**
+     * Load client configuration from a JSON file in the resource folder 
+     * 
+     * @param configFile This is the name of the JSON configuration file 
+     * @throws ParseException 
+     * @throws IOException 
+     */
+    public void loadConfig(String configFile) throws VIPClientInitException {
+    	ClassLoader classloader = Thread.currentThread().getContextClassLoader();
+    	InputStream is = classloader.getResourceAsStream(configFile + ".json");
+    	
+    	Reader reader = new InputStreamReader(is);
+    	JSONParser parser = new JSONParser();
+		JSONObject jsonObject = null;
+		try {
+			jsonObject = (JSONObject) parser.parse(reader);
+			this.vipServer = (String) jsonObject.get("online_service_url");
```

Comment:
to make it compatible, should it require to do same parsing for properties file's initialization?

Reply:
Support for backwards compatibility:

The new VIPCfg.loadConfig(String cfg) that will load the configuration from a JSON file will be available. However, when an existing application upgrades to the new version of the library, it doesn't have to use this new method. The old VIPCfg.initialize(String cfg) should still work, just marked with @Deprecated. 

Both methods will be available for quite some time until the deprecated method is no longer supported in some future version. In other words, we will remove VIPCfg.initialize(String cfg) eventually, but not immediately.

Reply:
fine.